profile/serializers.py

```diff
@@ -0,0 +1,144 @@
+from django.contrib.auth import authenticate
+from django.contrib.auth import get_user_model
+from django.utils.translation import ugettext_lazy as _
+from django.conf import settings
+from rest_framework import exceptions, serializers
+from allauth.utils import get_username_max_length, email_address_exists
+from allauth.account.adapter import get_adapter
+from allauth.account.utils import setup_user_email
+from allauth.account import app_settings as allauth_settings
+from .models import User
+
+UserModel = get_user_model()
+
+
+class UserSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = User
+        fields = ('pk', 'email', 'gender', 'is_superuser', 'password',)
+        extra_kwargs = {'password': {'write_only': True, 'min_length': 4}}
+
+
+class UserDetailsSerializer(serializers.ModelSerializer):
```

Comment:
Do you need a separate serializer to handle the details? Since the `password` field is write only on the serializer above, you should be able to use the same one both for deserialization and serialization.

Reply:
Well, extra_kwargs are not shown in the details serializer. This serializer accepts `password` as a parameter if the method is POST, UPDATE or PATCH but is not added to the serialized data in the GET method. Any way this API will be removed soon as dj-rest-auth was installed.